app/models/payment.rb

```diff
@@ -7,9 +7,11 @@ class Payment < ApplicationRecord
   validates :payroll_reference, :gross_value, :national_insurance, :employers_national_insurance, :tax, :net_pay, :gross_pay, presence: true, on: :upload
   validates :gross_value, :national_insurance, :employers_national_insurance, :student_loan_repayment, :tax, :net_pay, :gross_pay, numericality: true, allow_nil: true
 
-  delegate :first_name, :middle_name, :surname, :national_insurance_number, :payroll_gender, :date_of_birth, :email_address, :address_line_1, :address_line_2, :address_line_3, :address_line_4, :postcode, :has_student_loan, :student_loan_plan, :banking_name, :bank_sort_code, :bank_account_number, :building_society_roll_number, to: :claim
+  delegate :first_name, :middle_name, :surname, :national_insurance_number, :payroll_gender, :date_of_birth, :email_address, :address_line_1, :address_line_2, :address_line_3, :address_line_4, :postcode, :has_student_loan, :student_loan_plan, :banking_name, :bank_sort_code, :bank_account_number, :building_society_roll_number, to: :claim_for_personal_details
 
-  def claim
-    @claim ||= claims.first
+  private
+
+  def claim_for_personal_details
+    claims.first
```

Comment:
It seems like it does, although I don't fully understand why. Would you mind explaining how this caught your attention?

I'll restore the memoization.

Reply:
It caught my attention because I know how `ActiveRecord` translates things like `association.first` to a single select for efficiency instead of loading all the records to only return the one: https://apidock.com/rails/ActiveRecord/FinderMethods/first and because of a general spider sense for potential inefficiencies when I see code like `many_things.first`

Reply:
:+1:, thanks for the explanation.